js/index.ctrl.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+(function(){
+  angular
+  .module("radio")
+  .controller("RadioIndexCtrl", [
+    //"$sce",
+    "SongFactory",
+    ctrlFunc
+  ]);
+
+  function ctrlFunc(SongFactory){
+    this.songs = SongFactory.query();
+    this.showPlayer = false;
+    this.playerSong = null;
+
+    this.togglePlayer = function(songId){
+      if (this.showPlayer){this.showPlayer=false;}
+      else {
+        this.showPlayer = true;
+        this.playerSong = SongFactory.get({id: songId});
```

Comment:
We paired on this together, so feel confident that you have a better direction here now. Pretty much, instead of passing a `songId` and then using that to make another api call, we can just pass in the entire `song` object from the `index` view